front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -15,9 +15,71 @@ import type {
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
 import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+import type { ArrayElement } from 'utils/types';
 
 import { electricalProfilesDesignValues } from './consts';
 
+type SpeedLimitTagValue = ArrayElement<SimulationResponseSuccess['mrsp']['values']>;
+const getTag = (source?: SpeedLimitTagValue['source']): { name: string; color: string } => {
+  let name: string;
+  let color: string;
+
+  if (!source) {
+    name = 'missing_from_train';
+    color = '#94918E';
+  } else {
+    switch (source.speed_limit_source_type) {
+      case 'given_train_tag':
+        name = source.tag || 'missing_from_train';
```

Comment:
Not sure why we need the fallback here. Can `source.tag`equal an empty string ?

```suggestion
        name = source.tag || 'missing_from_train';
```

Reply:
You're right, I think we can just remove the fallback and only keep the source.tag